internal/pkg/deploy/deploy.go

```diff
@@ -87,6 +95,82 @@ func NewStore(sessProvider SessionProvider, store ConfigStoreClient) (*Store, er
 	return s, nil
 }
 
+// Pipeline is a deployed pipeline.
+type Pipeline struct {
+	appName      string
+	resourceName string
+	isLegacy     bool
+}
+
+// NewPipeline initiates a new pipeline. Note that this function should only be used for unit tests.
+func NewPipeline(appName, resourceName string, isLegacy bool) Pipeline {
+	return Pipeline{
+		appName:      appName,
+		resourceName: resourceName,
+		isLegacy:     isLegacy,
+	}
+}
+
+// HumanName returns the human-friendly name of the deployed pipeline.
+// For example: bad-goose (legacy)
+func (p Pipeline) HumanName() string {
+	if p.isLegacy {
+		return fmt.Sprintf("%s (legacy)", p.resourceName)
```

Comment:
What do you think of removing "legacy" and just having `Name`? I feel like it's going to spook users and force them to migrate for no reason

Reply:
Done!